File: leetcode/154-Find-Minimum-in-Rotated-Sorted-Array-II/phyllis.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {      
        int size = nums.size();        
        int prev = nums[0];
        
        if(size == 1) 
            return nums[0];
        
        for(int i = 1 ; i < size; i++){
            if(nums[i-1] > nums[i])
                return nums[i];                   
        }
        
        return nums[0];        
    }
};

File: leetcode/506-Relative-Ranks/phyllis.java
class Solution {
    public String[] findRelativeRanks(int[] nums) {
        Map<Integer, Integer> map = 
                        new TreeMap<Integer, Integer>();
        int length = nums.length;
        String[] ranks = new String[length];
        String[] medals = {"Gold Medal", "Silver Medal", "Bronze Medal"};
        for(int i = 0; i < length; i++){
             map.put(nums[i], i);
        }       
       
        int order = length;
        for(Integer key: map.keySet()) {
            int idx = map.get(key);          
            String strOrder = order <= 3 ? medals[order - 1] : String.valueOf(order); 
            ranks[idx] = String.valueOf(strOrder);
            order--;
        }
        
        return ranks;
    }
}
